io.openems.edge.controller.ess.cycle/src/io/openems/edge/controller/ess/cycle/ControllerEssCycleImpl.java

```diff
@@ -84,31 +92,62 @@ protected void deactivate() {
 
 	@Override
 	public void run() throws OpenemsNamedException {
-
-		// get max charge/discharge power
-		var maxDischargePower = this.ess.getPower().getMaxPower(this.ess, Phase.ALL, Pwr.ACTIVE);
-		var maxChargePower = this.ess.getPower().getMinPower(this.ess, Phase.ALL, Pwr.ACTIVE);
-
-		// store current state in StateMachine channel
-		var state = this.stateMachine.getCurrentState();
-		this.channel(ControllerEssCycle.ChannelId.STATE_MACHINE).setNextValue(state);
-
-		// Prepare Context
-		var previousState = this.stateMachine.getPreviousState();
-		var context = new Context(this, this.config, this.componentManager, this.ess, maxChargePower, maxDischargePower,
-				previousState, this.lastStateChange);
-
-		// Call the StateMachine
-		try {
-			this.stateMachine.run(context);
-		} catch (OpenemsNamedException e) {
-			this.logError(this.log, "StateMachine failed: " + e.getMessage());
+		if (this.parsedStartTime == null) {
+			this.logError(this.log, "Start time could not be parsed");
+			return;
 		}
-
-		if (previousState != state) {
-			// State has changed
-			this.lastStateChange = LocalDateTime.now(this.componentManager.getClock());
+		switch (this.config.mode()) {
+		case MANUAL_ON:
+			// get max charge/discharge power
+			var allowedDischargePower = this.ess.getPower().getMaxPower(this.ess, Phase.ALL, Pwr.ACTIVE);
+			var allowedChargePower = this.ess.getPower().getMinPower(this.ess, Phase.ALL, Pwr.ACTIVE);
```

Comment:
If you anyway provide `ess` to `Context`, then there is no need to read `allowedCharge/Dischare` here m.E.